modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,212 @@
+        for (int i = 0; i < detections.size(); ++i)
+        {
+            float* data = (float*)detections[i].data;
+            for (int j = 0; j < detections[i].total(); j += 7)
+            {
+                float conf = data[j + 2];
+                if (conf < confThreshold)
+                    continue;
+
+                float left   = data[j + 3];
+                float top    = data[j + 4];
+                float right  = data[j + 5];
+                float bottom = data[j + 6];
+                float width  = right  - left + 1;
+                float height = bottom - top  + 1;
+
+                if (absoluteCoords && width * height <= 1)
+                    boxes.emplace_back(left * frameWidth, top * frameHeight,
+                                       width * frameWidth, height * frameHeight);
+                else
+                    boxes.emplace_back(left, top, width, height);
+
+                classIds.push_back(static_cast<int>(data[j + 1]));
+                confidences.push_back(conf);
+            }
+        }
+    } else if (lastLayer->type == "Region") {
+        for (int i = 0; i < detections.size(); ++i) {
+            for (int j = 0; j < detections[i].rows; ++j) {
+
+                Mat scores = detections[i].row(j).colRange(5, detections[i].cols);
+                Point classIdPoint;
+                double conf;
+                minMaxLoc(scores, nullptr, &conf, nullptr, &classIdPoint);
+
+                if (conf < confThreshold)
+                    continue;
+
+                std::vector<Range> coord = {Range(j, j + 1), Range(0, 5)};
+                Mat coords;
+                detections[i](coord).copyTo(coords);
+
+                std::vector<float> bboxes = coords.reshape(1, 1);
+
+                float centerX = bboxes[0];
+                float centerY = bboxes[1];
+                float width   = bboxes[2];
+                float height  = bboxes[3];
+                float left    = centerX - width  / 2;
+                float top     = centerY - height / 2;
+
+                predClassIds.push_back(classIdPoint.x);
+                predConf.push_back(conf);
+
+                if (absoluteCoords)
+                    predBoxes.emplace_back(left * frameWidth, top * frameHeight,
+                                           width * frameWidth, height * frameHeight);
+                else
+                    predBoxes.emplace_back(left, top, width, height);
+            }
+        }
+    } else {
+        CV_Error(Error::StsError, "Unknown output layer type: \"" + lastLayer->type + "\"");
+    }
+
+    std::vector<int> indices;
+    NMSBoxes(predBoxes, predConf, confThreshold, nmsThreshold, indices);
+
+    boxes.reserve(indices.size());
+    confidences.reserve(indices.size());
+    classIds.reserve(indices.size());
+
+    for (int idx : indices)
+    {
+        boxes.push_back(predBoxes[idx]);
+        confidences.push_back(predConf[idx]);
+        classIds.push_back(predClassIds[idx]);
+    }
```

Comment:
We need to clean the vectors before or use `resize`.

Reply:
done